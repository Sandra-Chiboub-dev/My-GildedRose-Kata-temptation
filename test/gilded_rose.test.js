const {Shop, Item} = require("../src/gilded_rose");
const makeFixtureShop = () => {
  const items = [
    new Item("+5 Dexterity Vest", 10, 20),
    new Item("Aged Brie", 7, 46),
    new Item("Aged Brie", 2, 0),
    new Item("Elixir of the Mongoose", 5, 7),
    new Item("Sulfuras, Hand of Ragnaros", 0, 80),
    new Item("Sulfuras, Hand of Ragnaros", -1, 80),
    new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
    new Item("Backstage passes to a TAFKAL80ETC concert", 9, 52),
    new Item("Backstage passes to a TAFKAL80ETC concert", 9, 45),
    new Item("Backstage passes to a TAFKAL80ETC concert", 5, 43),
    new Item("Backstage passes to a TAFKAL80ETC concert", 5, 45),
    new Item("Conjured Mana Cake", 5, 18),
    new Item("Conjured Mana Cake", 3, 6),
    new Item("Conjured Mana Cake",-1, 6),
  ];

  return new Shop(items);
}

describe("Gilded Rose", () => {
  it("should make Fixture", () => {
    const actual = [];
    const days = 9;
    const shop = makeFixtureShop();

    for (let day = 0; day < days; day++) {
      actual.push(JSON.parse(JSON.stringify(shop.items)));
      shop.updateQuality();
    }
    expect(actual).toMatchSnapshot();
  });
});
