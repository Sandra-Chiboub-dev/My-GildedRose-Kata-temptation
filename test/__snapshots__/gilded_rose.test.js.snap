// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gilded Rose should make Fixture 1`] = `
Array [
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 46,
      "sellIn": 7,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 52,
      "sellIn": 9,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 45,
      "sellIn": 9,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 43,
      "sellIn": 5,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 45,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 18,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": -1,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 19,
      "sellIn": 9,
    },
    Object {
      "name": "Aged Brie",
      "quality": 47,
      "sellIn": 6,
    },
    Object {
      "name": "Aged Brie",
      "quality": 1,
      "sellIn": 1,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 6,
      "sellIn": 4,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 21,
      "sellIn": 14,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 54,
      "sellIn": 8,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 8,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 51,
      "sellIn": 4,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 53,
      "sellIn": 4,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 16,
      "sellIn": 4,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 4,
      "sellIn": 2,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 2,
      "sellIn": -2,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 18,
      "sellIn": 8,
    },
    Object {
      "name": "Aged Brie",
      "quality": 48,
      "sellIn": 5,
    },
    Object {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": 0,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 5,
      "sellIn": 3,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 22,
      "sellIn": 13,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 56,
      "sellIn": 7,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 52,
      "sellIn": 7,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 56,
      "sellIn": 3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 58,
      "sellIn": 3,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 14,
      "sellIn": 3,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 2,
      "sellIn": 1,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -2,
      "sellIn": -3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 17,
      "sellIn": 7,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 4,
    },
    Object {
      "name": "Aged Brie",
      "quality": 4,
      "sellIn": -1,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 4,
      "sellIn": 2,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 23,
      "sellIn": 12,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 58,
      "sellIn": 6,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 54,
      "sellIn": 6,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 61,
      "sellIn": 2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 63,
      "sellIn": 2,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 12,
      "sellIn": 2,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": 0,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -6,
      "sellIn": -4,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 16,
      "sellIn": 6,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 3,
    },
    Object {
      "name": "Aged Brie",
      "quality": 6,
      "sellIn": -2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 3,
      "sellIn": 1,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 24,
      "sellIn": 11,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 63,
      "sellIn": 5,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 59,
      "sellIn": 5,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 66,
      "sellIn": 1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 68,
      "sellIn": 1,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 10,
      "sellIn": 1,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -4,
      "sellIn": -1,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -10,
      "sellIn": -5,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 15,
      "sellIn": 5,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 8,
      "sellIn": -3,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 2,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 27,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 68,
      "sellIn": 4,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 64,
      "sellIn": 4,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 71,
      "sellIn": 0,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 73,
      "sellIn": 0,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 8,
      "sellIn": 0,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -8,
      "sellIn": -2,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -14,
      "sellIn": -6,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 14,
      "sellIn": 4,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 1,
    },
    Object {
      "name": "Aged Brie",
      "quality": 10,
      "sellIn": -4,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -1,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 31,
      "sellIn": 9,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 73,
      "sellIn": 3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 69,
      "sellIn": 3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 5,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 5,
      "sellIn": -1,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 4,
      "sellIn": -1,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -12,
      "sellIn": -3,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -18,
      "sellIn": -7,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 13,
      "sellIn": 3,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 0,
    },
    Object {
      "name": "Aged Brie",
      "quality": 12,
      "sellIn": -5,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -2,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 35,
      "sellIn": 8,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 78,
      "sellIn": 2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 74,
      "sellIn": 2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 5,
      "sellIn": -2,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 5,
      "sellIn": -2,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 0,
      "sellIn": -2,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -16,
      "sellIn": -4,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -22,
      "sellIn": -8,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 12,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -1,
    },
    Object {
      "name": "Aged Brie",
      "quality": 14,
      "sellIn": -6,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 0,
      "sellIn": -3,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 39,
      "sellIn": 7,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 83,
      "sellIn": 1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 79,
      "sellIn": 1,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 5,
      "sellIn": -3,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 5,
      "sellIn": -3,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -4,
      "sellIn": -3,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -20,
      "sellIn": -5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": -26,
      "sellIn": -9,
    },
  ],
]
`;
